mmaction/datasets/audio_dataset.py

```diff
@@ -0,0 +1,135 @@
+                idx = 0
+                filename = line_split[idx]
+                if self.data_prefix is not None:
+                    if not filename.endswith(self.suffix):
+                        filename = osp.join(self.data_prefix,
+                                            filename) + self.suffix
+                    else:
+                        filename = osp.join(self.data_prefix, filename)
+                video_info['audio_path'] = filename
+                idx += 1
+                # idx for total_frames
+                video_info['total_frames'] = int(line_split[idx])
+                idx += 1
+                # idx for label[s]
+                label = [int(x) for x in line_split[idx:]]
+                assert len(label), f'missing label in line: {line}'
+                if self.multi_class:
+                    assert self.num_classes is not None
+                    onehot = torch.zeros(self.num_classes)
+                    onehot[label] = 1.0
+                    video_info['label'] = onehot
+                else:
+                    assert len(label) == 1
+                    video_info['label'] = label[0]
+                video_infos.append(video_info)
+
+        return video_infos
+
+    def evaluate(self,
+                 results,
+                 metrics='top_k_accuracy',
+                 topk=(1, 5),
+                 logger=None):
+        """Evaluation in rawframe dataset.
+
+        Args:
+            results (list): Output results.
+            metrics (str | sequence[str]): Metrics to be performed.
+                Defaults: 'top_k_accuracy'.
+            logger (obj): Training logger. Defaults: None.
+            topk (tuple[int]): K value for top_k_accuracy metric.
+                Defaults: (1, 5).
+            logger (logging.Logger | None): Logger for recording.
+                Default: None.
+
+        Return:
+            dict: Evaluation results dict.
+        """
+        if not isinstance(results, list):
+            raise TypeError(f'results must be a list, but got {type(results)}')
+        assert len(results) == len(self), (
+            f'The length of results is not equal to the dataset len: '
+            f'{len(results)} != {len(self)}')
+
+        if not isinstance(topk, (int, tuple)):
+            raise TypeError(
+                f'topk must be int or tuple of int, but got {type(topk)}')
+
+        metrics = metrics if isinstance(metrics, (list, tuple)) else [metrics]
+        allowed_metrics = ['top_k_accuracy', 'mean_class_accuracy']
+        for metric in metrics:
+            if metric not in allowed_metrics:
+                raise KeyError(f'metric {metric} is not supported')
+
+        eval_results = {}
+        gt_labels = [ann['label'] for ann in self.video_infos]
+
+        for metric in metrics:
+            msg = f'Evaluating {metric}...'
+            if logger is None:
+                msg = '\n' + msg
+            print_log(msg, logger=logger)
```

Comment:
there is a logger in the argument list, right? how about making that one defaults to printlog

Reply:
Agree, I think it is better to construct the var `msg` and print it once at the final lines rather than print it several times; And this `evaluate` is exactly the same for all datasets, so redundant.